node/network/src/proxy/mod.rs

```diff
@@ -0,0 +1,71 @@
+use std::sync::Arc;
+
+use futures::stream::Stream;
+use futures::sync::mpsc::{channel, Receiver, Sender};
+
+use primitives::serialize::Encode;
+
+use crate::peer::PeerMessage;
+use crate::protocol::{forward_msg, PackedMessage};
+
+pub mod dropout;
+
+pub struct Proxy {
+    handlers: Vec<Arc<ProxyHandler>>,
+}
+
+/// All messages before being sent to other participants will be processed by several handlers
+/// chained sequentially in a pipeline.
+///
+/// Each handler will receive an iterator of messages and will return (not necessarily the same)
+/// iterator of messages. This method allows easy manipulation of messages before being emitted
+/// to other authorities through the network.
+///
+/// The order in which handlers are grouped can affect the behavior of the proxy.
+impl Proxy {
+    pub fn new(handlers: Vec<Arc<ProxyHandler>>) -> Self {
```

Comment:
`Proxy` should have a check that whenever there are no handlers it uses the old `forward_msg` and `forward_msgs` code to send messages.

Reply:
The current implementation works this way (only use `forward_msg`). When `self.handlers` is empty only one task is spawned at `send_final_message`.